### ISSUE TITLE
Complex Document Type in DocTypeGrid

### ISSUE THREAD

**Issue by aDotNetDeveloper** [KevinJump/uSync3.0#114]
Hi Kevin

currently using uSync 3.3.1.740 on Umb 7.5.4

When an array like property (Imulus.UrlPicker or Umbraco.RadioButtonList) is on a Document Type that is within a DocTypeGrid, this exception occurs on export;
```
Jumoo.uSync.Content.ContentHandler - Error saving Content: System.InvalidCastException: Unable to cast object of type 'Newtonsoft.Json.Linq.JArray' to type 'Newtonsoft.Json.Linq.JObject'.
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
   at Jumoo.uSync.ContentMappers.DocTypeGridMapper.ProcessDocTypeGridValues(String content, Boolean import)
   at Jumoo.uSync.ContentMappers.DocTypeGridMapper.GetExportValue(Int32 dataTypeDefinitionId, String value)
   at Jumoo.uSync.Core.Mappers.GridMapper.ProcessGridValues(String content, Boolean import)
   at Jumoo.uSync.Core.Mappers.GridMapper.GetExportValue(Int32 dataTypeDefinitionId, String value)
```
This is the _value_ DocTypeGridMapper.GetExportValue() is getting - which is valid for a Imulus.UrlPicker
```[
  {
    "type": "content",
    "meta": {
      "title": "About",
      "newWindow": false
    },
    "typeData": {
      "url": "",
      "contentId": 1154,
      "mediaId": null
    },
    "disabled": false
  }
]
```
For Imulus.UrlPicker I'm using a custom mapper which verifies that the array contains 'UrlPicker.Umbraco.Models.UrlPicker' objects, using the following code;
````
    var array = JsonConvert.DeserializeObject<Newtonsoft.Json.Linq.JArray>(value);
    foreach (var obj in array)
    {
        var x = JsonConvert.DeserializeObject<UrlPicker.Umbraco.Models.UrlPicker>(obj.ToString());
    }
    return JsonConvert.SerializeObject(array);
````
Regardless, the DocTypeGrid mapper still fails on this and on Umbraco.RadioButtonList which is using standard mapping.

I should also mention that the Grid is also within a Vorto.

Hope you can shed some light on what can be done to get some form of solution.

regards
Wayne

**Comment by KevinJump** [comment]
thanks, I will take a look

**Comment by KevinJump** [comment]
I have tested and I think this is fixed in the latest release 3.3.2
